### PR TITLE
feat(tx-interpret): sign transaction interpretation

### PR DESCRIPTION
**Pull request by superokid**
## Changes description
When signing from dapps, should show transaction interpretation. Users should understand what they signing, for example, sending fund before they approve the transaction

## Preview
- Ref finance swap
Screenshot 2024-08-07 at 15 45 00

- Ref finance add liquidity
Screenshot 2024-08-06 at 18 23 40

- Ref finance stake liquidity pool
Screenshot 2024-08-06 at 16 23 24

- Burrow supply
Screenshot 2024-08-02 at 16 33 41

- Burrow staking
Screenshot 2024-08-08 at 12 13 40

- Burrow withdraw
Screenshot 2024-08-08 at 12 13 40

- Paras buy nft
Screenshot 2024-08-02 at 18 34 00

- Linear staking
Screenshot 2024-08-07 at 14 10 18

- Linear liquid unstake
Screenshot 2024-08-07 at 14 08 05

### CODE REVIEW

**Comment by superokid** [comment]
@race-of-sloths

**Comment by race-of-sloths** [comment]
@superokid Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!


[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/superokid?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/superokid?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/superokid)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/superokid)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by edwardchew97** [comment]
@race-of-sloths score 8

**Review by edwardchew97 — APPROVED** [review]
Seems good and approving it since it doesn't involve any changes in transaction signing process.
This will require more testing before being shipped on production. We will however need to comment out transaction executor and enable it only later to make sure things are better tested and progressively ship new things.

**Pull request merged**
